src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [ separator, ...oldFormat ] = fromFormat.reverse();
+  const newSeparator = toFormat[toFormat.length - 1];
+  const arrayDate = date.split(separator).reverse();
+  let year = '';
+  const oldFormatDate = {};
+  let result = '';
+
+  for (let index = 0; index < oldFormat.length; index++) {
+    if (oldFormat[index].includes('Y')) {
+      year = arrayDate[index];
+    }
+
+    oldFormatDate[oldFormat[index]] = arrayDate[index];
+  }
+
+  for (let index = 0; index < toFormat.length - 1; index++) {
+    if (toFormat[index].includes('Y')) {
+      const newYear = convertingYear(toFormat[index], year);
+
+      result += `${newYear}`;
+    } else {
+      const partDate = oldFormatDate[toFormat[index]];
+
+      result += `${partDate}`;
+    }
+
+    if (index !== toFormat.length - 2) {
+      result += newSeparator;
+    }
+  }
+
+  return result;
+}
+
+function convertingYear(newFormat, year) {
```

Comment:
Function names should be in camelCase. Rename 'convertingYear' to 'convertYear'.